### PR TITLE
refactor(HOC): extract authenticate functions

### PR DESCRIPTION
**Pull request by Frikki**
## Manditory Checks
- [ ] I added new tests for the issue I fixed or the feature I built
- [x] I ran `npm test` for the package I'm modifying
- [ ] I used `npm run commit` instead of `git commit`

---

Extract authenticate functions.
Rename types and interfaces.

### CODE REVIEW

**Comment by Frikki** [comment]
Please go through this carefully @TylorS

File: src/drivers/firebase-authentication/createUserCredential$.ts [hunk 1]
```diff
@@ -2,15 +2,15 @@ import { Stream, just } from 'most';
...
+  AuthenticationType, CreateUserAuthentication,
+  EmailAndPasswordAuthentication, PopupAuthentication,
   REDIRECT, EMAIL_AND_PASSWORD, CREATE_USER, POPUP, ANONYMOUSLY, SIGN_OUT,
   GET_REDIRECT_RESULT
 } from './types';
 import { convertUserToUserCredential } from './convertUserToUserCredential';
 import { defaultUserCredential } from './defaultUserCredential';
 
-export function createUserCredential$(method: string, authenticationInput: AuthenticationInput, firebaseInstance: any):
+export function createUserCredential$(method: string, authenticationInput: AuthenticationType, firebaseInstance: any):
```
Comment (TylorS) [inline src/drivers/firebase-authentication/createUserCredential$.ts]: rename `authenticationInput` to `authenticationType`

File: src/drivers/firebase-authentication/createUserCredential$.ts [hunk 2]
```diff
@@ -55,23 +55,23 @@ export function createUserCredential$(method: string, authenticationInput: Authe
   return just(defaultUserCredential);
 }
 
-function emailAndPasswordSignIn(authenticationInput: AuthenticationInput, firebaseInstance: any) {
-  const { email, password } = (authenticationInput as EmailAndPasswordAuthenticationInput);
+function emailAndPasswordSignIn(authenticationInput: AuthenticationType, firebaseInstance: any) {
```
Comment (TylorS) [inline src/drivers/firebase-authentication/createUserCredential$.ts:60]: Rename `authenticationInput` to `authenticationType`

File: src/drivers/firebase-authentication/createUserCredential$.ts [hunk 3]
```diff
@@ -55,23 +55,23 @@ export function createUserCredential$(method: string, authenticationInput: Authe
...
+  const { email, password } = (authenticationInput as EmailAndPasswordAuthentication);
 
   return fromFirebasePromise<firebase.User>(
     firebaseInstance.auth().signInWithEmailAndPassword(email, password))
     .map(convertUserToUserCredential(new firebase.auth.EmailAuthProvider()));
 }
 
-function popupSignIn(authenticationInput: AuthenticationInput, firebaseInstance: any) {
-  const { provider } = authenticationInput as PopupAuthenticationInput;
+function popupSignIn(authenticationInput: AuthenticationType, firebaseInstance: any) {
```
Comment (TylorS) [inline src/drivers/firebase-authentication/createUserCredential$.ts:68]: same here

File: src/drivers/firebase-authentication/createUserCredential$.ts [hunk 4]
```diff
@@ -55,23 +55,23 @@ export function createUserCredential$(method: string, authenticationInput: Authe
...
+  const { provider } = authenticationInput as PopupAuthentication;
 
   return fromFirebasePromise<firebase.auth.UserCredential>(
     firebaseInstance.auth().signInWithPopup(provider));
 }
 
-function redirectSignIn(authenticationInput: AuthenticationInput, firebaseInstance: any) {
-  const { provider } = authenticationInput as PopupAuthenticationInput;
+function redirectSignIn(authenticationInput: AuthenticationType, firebaseInstance: any) {
```
Comment (TylorS) [inline src/drivers/firebase-authentication/createUserCredential$.ts:75]: and again here

File: src/higher-order-components/authenticate/authenticate.test.ts [hunk 5]
```diff
@@ -61,7 +61,7 @@ describe('higher-order-components/authenticate', () => {
         it('should contain value of type AuthenticationInput', () => {
           const { authentication$ } = Component(defaultSources);
 
-          return authentication$.observe((authInput: AuthenticationInput) => {
+          return authentication$.observe((authInput: AuthenticationType) => {
```
Comment (TylorS) [inline src/higher-order-components/authenticate/authenticate.test.ts]: authInput -> authenticationType

File: src/higher-order-components/authenticate/model.test.ts [hunk 6]
```diff
@@ -17,7 +17,7 @@ describe('authenticate', () => {
     });
 
     it('should output a state object', () => {
-      const authenticationInput: AuthenticationInput = model({ method: GOOGLE });
+      const authenticationInput: AuthenticationType = model({ method: GOOGLE });
```
Comment (TylorS) [inline src/higher-order-components/authenticate/model.test.ts]: authenticationInput -> authenticationType

**Review by TylorS — CHANGES_REQUESTED** [review]
Really just think that parameters and variables named `authenticationInput` should be renamed to match the new `AuthenticationType` interface.

File: src/drivers/firebase-authentication/createUserCredential$.ts (hunk 1 above)
Comment (Frikki) [reply src/drivers/firebase-authentication/createUserCredential$.ts]: ✅

File: src/drivers/firebase-authentication/createUserCredential$.ts (hunk 2 above)
Comment (Frikki) [reply src/drivers/firebase-authentication/createUserCredential$.ts:60]: ✅

File: src/drivers/firebase-authentication/createUserCredential$.ts (hunk 3 above)
Comment (Frikki) [reply src/drivers/firebase-authentication/createUserCredential$.ts:68]: ✅

File: src/drivers/firebase-authentication/createUserCredential$.ts (hunk 4 above)
Comment (Frikki) [reply src/drivers/firebase-authentication/createUserCredential$.ts:75]: ✅

File: src/higher-order-components/authenticate/authenticate.test.ts (hunk 5 above)
Comment (Frikki) [reply src/higher-order-components/authenticate/authenticate.test.ts]: ✅

File: src/higher-order-components/authenticate/model.test.ts (hunk 6 above)
Comment (Frikki) [reply src/higher-order-components/authenticate/model.test.ts]: ✅

**Comment by Frikki** [comment]
It’s best to rebase the PR before merging @TylorS

**Review by TylorS — APPROVED** [review]

**Pull request merged**